src/sentry/integrations/slack/views/unlink_team.py

```diff
@@ -0,0 +1,116 @@
+)
+from . import build_linking_url as base_build_linking_url
+from . import never_cache
+from .link_team import INSUFFICIENT_ROLE_MESSAGE, INSUFFICIENT_ROLE_TITLE
+
+SUCCESS_UNLINKED_TITLE = "Team unlinked"
+SUCCESS_UNLINKED_MESSAGE = (
+    "This channel will no longer receive issue alert notifications for the {team} team."
+)
+
+
+def build_team_unlinking_url(
+    integration: Integration,
+    organization_id: str,
+    slack_id: str,
+    channel_id: str,
+    channel_name: str,
+    response_url: str,
+) -> str:
+    return base_build_linking_url(
+        "sentry-integration-slack-unlink-team",
+        integration_id=integration.id,
+        organization_id=organization_id,
+        slack_id=slack_id,
+        channel_name=channel_name,
+        channel_id=channel_id,
+        response_url=response_url,
+    )
+
+
+class SlackUnlinkTeamView(BaseView):  # type: ignore
+    @transaction_start("SlackUnlinkIdentityView")
+    @never_cache
+    def handle(self, request: Request, signed_params: str) -> Response:
+        params = unsign(signed_params)
+
+        organization, integration, idp = get_identity(
+            request.user, params["organization_id"], params["integration_id"]
+        )
+        channel_name = params["channel_name"]
+        channel_id = params["channel_id"]
+
+        try:
+            external_team = ExternalActor.objects.get(
+                organization=organization,
+                integration=integration,
+                provider=ExternalProviders.SLACK.value,
+                external_name=channel_name,
+                external_id=channel_id,
+            )
+        except IntegrityError as e:
+            logger.error("slack.team.unlink.integrity-error", extra=e)
+            raise Http404
+
+        team = Team.objects.get(actor=external_team.actor)
+        if request.method != "POST":
+            return render_to_response(
+                "sentry/integrations/slack-unlink-team.html",
+                request=request,
+                context={
+                    "team": team,
+                    "channel_name": channel_name,
+                    "provider": integration.get_provider(),
+                },
+            )
+
+        org_member = get_org_member_by_slack_id(
+            request, integration, organization, params["slack_id"]
+        )
+        if not is_valid_role(org_member, team, organization):
+            return send_confirmation(
+                integration,
+                channel_id,
+                INSUFFICIENT_ROLE_TITLE,
+                INSUFFICIENT_ROLE_MESSAGE,
+                "sentry/integrations/slack-post-linked-team.html",
+                request,
+            )
+
+        external_team.delete()
+
+        # Turn off notifications for all of a team's projects.
+        NotificationSetting.objects.update_settings(
+            ExternalProviders.SLACK,
+            NotificationSettingTypes.ISSUE_ALERTS,
+            NotificationSettingOptionValues.NEVER,
```

Comment:
It would be better to just wipe out all notification settings for the team. If we add project-specific settings they'll still override this.
```python
NotificationSetting.objects.remove_for_team(team)
```